include/rx/ranges.hpp

```diff
@@ -1807,6 +1807,33 @@ struct none_of {
 };
 template <class P>
 none_of(P &&)->none_of<P>;
+    
+template <class P>
+struct contains {
+    const P& pred;
```

Comment:
I think it's better to hold `P` by value, and take it by value in the constructor.

Reply:
Passing by value may be expensive. I tested it against a structure to see if the reference won't dangle and it doesn't because contains is always called at the very end.

Reply:
It will dangle if captured by rvalue and stored in local variable.

The following is allowed:

```
auto f = transform(...) | contains(some_expression);
bool success = input | f;
```

In this case, the return value of `some_expression` will not be lifetime-extended, and the internal reference will be dangling.

Do you have a use case in mind where moving the value into `compare()` would be expensive?